src/commonMain/kotlin/org/kosat/Kosat.kt

```diff
@@ -2,12 +2,23 @@ package org.kosat
 
 import kotlin.math.abs
 
-class Kosat(clauses: MutableList<MutableList<Lit>>, vars: Int = 0): Solver {
+fun solveWithAssumptions(cnf: CnfRequest): List<Int>? {
```

Comment:
Why two methods with equivalent signatures? If `CnfRequest` already "knows" whether it has assumptions or not, why bother *yourself* (a user) with two "different" (but similar) methods?